crates/bytesbuf/benches/bytesbuf_vs_bytes.rs

```diff
@@ -0,0 +1,1274 @@
+            for _ in 0..iters {
+                let mut sb = BytesBuf::new();
+                sb.reserve(COPY_TO_SLICE_LEN, &transparent_memory);
+                let data = [0xCD_u8; COPY_TO_SLICE_LEN];
+                sb.put_slice(&data[..]);
+                black_box(sb);
+            }
+            start.elapsed()
+        });
+    });
+
+    // put_bytes (BytesView) - unique to bytesbuf, no bytes equivalent
+    let allocs_op = allocs.operation("put_bytes_view");
+    group.bench_function("put_bytes_view", |b| {
+        b.iter_custom(|iters| {
+            let _span = allocs_op.measure_thread().iterations(iters);
+            let start = Instant::now();
+            for _ in 0..iters {
+                let mut sb = BytesBuf::new();
+                sb.put_bytes(test_data_as_seq.clone());
+                black_box(sb);
+            }
+            start.elapsed()
+        });
+    });
+
+    // put_byte (bytesbuf) vs put_u8 (bytes)
+    let allocs_op = allocs.operation("put_byte");
+    group.bench_function("put_byte", |b| {
+        b.iter_custom(|iters| {
+            let _span = allocs_op.measure_thread().iterations(iters);
+            let start = Instant::now();
+            for _ in 0..iters {
+                let mut sb = BytesBuf::new();
+                sb.reserve(1, &transparent_memory);
+                sb.put_byte(black_box(0xAB));
+                black_box(sb);
+            }
+            start.elapsed()
+        });
+    });
+
+    let allocs_op = allocs.operation("put_u8");
+    group.bench_function("put_u8", |b| {
+        b.iter_custom(|iters| {
+            let _span = allocs_op.measure_thread().iterations(iters);
+            let start = Instant::now();
+            for _ in 0..iters {
+                let mut sb = BytesBuf::new();
+                sb.reserve(1, &transparent_memory);
+                sb.put_u8(black_box(0xAB));
+                black_box(sb);
+            }
+            start.elapsed()
+        });
+    });
+
+    // put_byte_repeated (bytesbuf) vs put_bytes (bytes) - these are equivalent!
+    let allocs_op = allocs.operation("put_byte_repeated");
+    group.bench_function("put_byte_repeated", |b| {
+        b.iter_custom(|iters| {
+            let _span = allocs_op.measure_thread().iterations(iters);
+            let start = Instant::now();
+            for _ in 0..iters {
+                let mut sb = BytesBuf::new();
+                sb.reserve(COPY_TO_SLICE_LEN, &transparent_memory);
+                sb.put_byte_repeated(black_box(0xCD), COPY_TO_SLICE_LEN);
+                black_box(sb);
+            }
+            start.elapsed()
+        });
+    });
+
+    let allocs_op = allocs.operation("put_bytes");
+    group.bench_function("put_bytes", |b| {
+        b.iter_custom(|iters| {
+            let _span = allocs_op.measure_thread().iterations(iters);
+            let start = Instant::now();
+            for _ in 0..iters {
+                let mut sb = BytesBuf::new();
```

Comment:
Only the function we are benchmarking should be part of the timed loop. Any data preparation needs to happen before this loop (e.g. by creating a second loop before we start the measurements).